### PR TITLE
Build static lib too

### PR DESCRIPTION
**Pull request by davidbrochart**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

@conda-forge-admin, please rerender

### CODE REVIEW

**Review by xhochy — CHANGES_REQUESTED** [review]
Please separate static libs into a separate package

File: recipe/install.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+if [[ "$PKG_NAME" == *static ]]
+then
+    cd build-static
+    ninja install
+    # relying on conda to dedup package
+    echo "Keeping all files, conda will dedupe"
+else
+    cd build-shared
+    ninja install
+    rm -rf ${PREFIX}/lib/*.a
```
Comment (xhochy) [inline recipe/install.sh:10]: Is this really needed?

File: recipe/bld.bat [hunk 2]
```diff
@@ -12,3 +12,19 @@ if errorlevel 1 exit 1
 
 ninja install
```
Comment (xhochy) [inline recipe/bld.bat:13]:
```suggestion
ninja
```

File: recipe/bld.bat [hunk 3]
```diff
@@ -12,3 +12,19 @@ if errorlevel 1 exit 1
...
+cmake -G "Ninja" ^
+      -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
+      -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^
+      -DCMAKE_INSTALL_LIBDIR=lib ^
+      -DCMAKE_BUILD_TYPE=Release ^
+      -DBUILD_SHARED_LIBS=OFF ^
+      ..
+if errorlevel 1 exit 1
+
+ninja install
```
Comment (xhochy) [inline recipe/bld.bat:29]:
```suggestion
ninja
```

**Comment by davidbrochart** [comment]
I don't understand what should be in the outputs requirements and what should be in the requirements outside the outputs. Here `libopenssl-static` should provide `LibCrypto` but we have:
```
CMake Error at /home/conda/feedstock_root/build_artifacts/s2n_1605621174836/_build_env/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find LibCrypto (missing: LibCrypto_LIBRARY)
```

**Comment by davidbrochart** [comment]
@conda-forge-admin, please rerender

**Comment by davidbrochart** [comment]
Now that Windows is enabled, it's missing `LibCrypto`, but not for Unix:
```
CMake Error at D:/bld/s2n_1605629367149/_build_env/Library/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find LibCrypto (missing: LibCrypto_LIBRARY)
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -22,11 +21,42 @@ requirements:
...
+    script: install.sh  # [unix]
+    script: install.bat  # [win]
+    requirements:
+      build:
+        - cmake
+        - {{ compiler('c') }}
+        - ninja
+      host:
+        - openssl
+        - libopenssl-static
```
Comment (isuruf) [inline recipe/meta.yaml:52]: pin `s2n` here exactly.

Comment (davidbrochart) [reply recipe/meta.yaml:52]: Now linux fails, but I'm not sure it's the right way to pin `s2n`.

**Pull request merged**